dotnet-desktop-guide/net/wpf/events/routed-events-overview.md

```diff
@@ -0,0 +1,260 @@
+
+:::code language="xaml" source="./snippets/routed-events-overview/csharp/MainWindow.xaml" id="AddHandler_QualifiedEventName":::
+
+The signature of the event handler method in code-behind must match the delegate type for the routed event. The `sender` parameter of the <xref:System.Windows.RoutedEventHandler> delegate for the <xref:System.Windows.Controls.Primitives.ButtonBase.Click> event specifies the element to which the event handler is attached. The `args` parameter of the `RoutedEventHandler` delegate contains the event data. A compatible code-behind implementation for the `Button_Click` event handler might be:
+
+:::code language="csharp" source="./snippets/routed-events-overview/csharp/MainWindow.xaml.cs" id="ButtonClickHandler":::
+:::code language="vb" source="./snippets/routed-events-overview/vb/MainWindow.xaml.vb" id="ButtonClickHandler":::
+
+Although <xref:System.Windows.RoutedEventHandler> is the basic routed event handler delegate, some controls or implementation scenarios require different delegates that support more specialized event data. As an example, for the <xref:System.Windows.UIElement.DragEnter> routed event, your handler should implement the <xref:System.Windows.DragEventHandler> delegate. By doing so, your handler code can access the <xref:System.Windows.DragEventArgs.Data?displayProperty=nameWithType> property in event data, which contains the clipboard payload from the drag operation.
+
+For a complete example of how to attach an event handler to an element using XAML, see [How to handle a routed event](/dotnet/desktop/wpf/advanced/how-to-handle-a-routed-event?view=netframeworkdesktop-4.8&preserve-view=true).
+
+To attach an event handler for a routed event to an element using code, you generally have two options:
+
+- Directly call the <xref:System.Windows.UIElement.AddHandler%2A> method. Routed event handlers can always be attached this way. This example attaches a `Click` event handler to a button using the `AddHandler` method:
+
+  :::code language="csharp" source="./snippets/routed-events-overview/csharp/MainWindow.xaml.cs" id="AddHandler_ToAttachHandler":::
+  :::code language="vb" source="./snippets/routed-events-overview/vb/MainWindow.xaml.vb" id="AddHandler_ToAttachHandler":::
+
+- If the routed event implements a CLR event wrapper, use language-specific event syntax to add event handlers just as you would for a standard CLR event. Most existing WPF routed events implement the CLR wrapper, thus enabling language-specific event syntax. This example attaches a `Click` event handler to a button using language specific syntax:
+
+  :::code language="csharp" source="./snippets/routed-events-overview/csharp/MainWindow.xaml.cs" id="LanguageSpecificSyntax_ToAttachHandler":::
+  :::code language="vb" source="./snippets/routed-events-overview/vb/MainWindow.xaml.vb" id="LanguageSpecificSyntax_ToAttachHandler":::
```

Comment:
I know this is a bit more involved than just scrubbing the article, but I feel like it may be more beneficial for our customers to see this implemented as the StackPanel handling the button click instead of a button handling the button click. The demonstration doesn't seem to show anything about the "routed" aspect.

Reply:
The existing AddHandler example mirrors the CLR event example. I've added the routed event scenario you mentioned (attach to StackPanel) as an additional example.